### DISCUSSION TITLE
Individual Meeting vs Group Radio button + Taxonomy implications

### DISCUSSION THREAD

**Discussion by seanalansnyder** [code4recovery/12-step-meeting-list#1349] [General]
I'm trying to sort understand the logic behind the tsml_group custom post type, and how it relates and/or is different from the tsml_meeting post type within functions.php.   And, relatedly, the logic 
behind hiding the tsml_district (Taxonomy) behind the Individual Meeting/Part of a Group Radio button in admin_meeting.php.
Can anyone shed some light?

In my mind, an individual Meeting (or even a Location) post_type should be able to be associated with a tsml_district Taxonomy term, no?     (You wouldn't believe how much hair I pulled out trying to figure out how to assign a District to a Meeting in Admin GUI...)    

Ultimately, where I'm going is that I'd like to be able to build-out and leverage that Districts (tsml_district) taxonomy for other post types within the site I'm working on (beyond TSML).  

Example:  
I'm building out an Events page using Modern Events Calendar Lite as the back-end.  

- I'd love to make the tsml_district taxonomy public, and apply to mec-events custom post types.
- I'd love to be able to access (make public) tsml_location post_type, and leverage that information (cross-reference) within the mec_location Taxonomy.
- I'd love to leverage tsml_meeting / tsml_group post_type's data, and cross reference it to mec_organizer Taxonomy.

My inner data scientist bristles at the thought of all of the redundant data entry involved when the FriendsOfBill Meeting wants to post a ~~Flyer~~ EVENT (with usable iCal .ics file AND location metadata) for their 3rd Anniversary OpenTalk & Chili Cook-Off.   95% of our Event Organizers are already in TSML as a tsml_meeting or tsml_group.   95% of our Event Locations are already in TSML  as tsml_locations.    Waste time doing double-entry?

From a usability perspective, It'd be super swell to have a District 12 Page with Query Loop sections for tsml_meetings, posts, and mec-events all filtered with tsml_district=12.   Ditto for an RSS category feed by District.   I may be reaching here, but I'd love to start typing "FriendsOf..." when filling out the Organizer field on a new event, and have it autocomplete from tsml_meeting data.

I believe I can do MOST of what I'm looking to do by adding additional Relationship field with the Pods plug-in  (very similar to ACF that was mentioned in another discussion).   I guess I'm just looking to see if/how others have tackled this duplication of data / cross-reference issue?

**Comment by seanalansnyder** [comment]
Related Side note:
Is there a specific reason we didn't add the 'labels' array for custom post types tsml_location and tsml_group when registering them, beyond the fact that they don't seem intended to be publicly visible?

The ramifications of the lack of a name and singular_name labels for these post types means we have 4 different post types that are named "Posts" when viewed by other plug-ins.

Pods:
![image](https://github.com/code4recovery/12-step-meeting-list/assets/31214020/280b872b-62d8-4b97-98c9-a729c56fbcc7)

Tools > Export:
![image](https://github.com/code4recovery/12-step-meeting-list/assets/31214020/c2d5041a-377a-4d92-9751-5b609f097dca)

**Comment by joshreisner** [comment]
hi @seanalansnyder we'd be happy to add the labels array for tsml_locations and tsml_groups - we didn't bother before because there is no context within the plugin where you'd need them, but it won't add much complexity for us to add a few labels so it will work with your calendar plugin. i'm happy to raise a PR for that unless you would like to?

as for the other question, locations have regions and groups have districts - i don't think we're looking to change that. if you want to associate events with groups that feels like it would make sense to me. you can always create groups for just one meeting if you need to. if you create a custom district wordpress taxonomy archive i think you can probably customize that to display all the groups and meetings using our `tsml_get_meetings(['district' => 'district-12'])` function…
